### ISSUE TITLE
Can't access state when use it with hooks component

### ISSUE THREAD

**Issue by armspkt** [sarneeh/reaptcha#194]
```
const [email, setEmail] = useState('')

const onVerify = response => {
  console.log(email) ??
}

```

Email does not update state to the latest.
when I set email state to 'xxx' (setEmail('xxx'))

but outside onVerify function can access email state 
console.log(email) // 'xxx'

Can I use reapctcha with hook component?

**Comment by stfenjobs** [comment]
me too , i have to use  https://github.com/dozoisch/react-google-recaptcha

**Comment by dizefurkan** [comment]
Same issue.
I change my Functional Component to Class Component because of this hooks state problem.
Now it's working.

By the way, i used before react-google-recaptcha on our multiple route. like login/register/forget-password.
Problem that library: When route change, (on componentUnmount) recaptcha not removed from dom. That's why i use that library. in appearance this library solve that issue :) In my opinion it's more important the class/functional component issue

**Comment by arya-s** [comment]
I'm not sure why you'd want to read out state in `onVerify`. All you need to do is to switch a flag that represents being verified.

e.g.
```
const [verified, setVerified] = useState(false)
const [email, setEmail] = useState('')

const verify = () => setVerified(true)
const expire = () => setVerified(false)

  ...

const handleSubmit = (e) => {  if (verified) sendEmail(email) }

return (
   <>
      <Reaptcha sitekey="..." onVerify={verify} onError={expire} onExpire={expire} />
     <form onSubmit={handleSubmit>

        ...

        <button type="submit" disabled={!verified}>Submit</button>
     </form>
   </>
)
```

**Comment by cheesestringer** [comment]
> I'm not sure why you'd want to read out state in `onVerify`. All you need to do is to switch a flag that represents being verified.

The token needs to be submitted to your server to be verified.
See https://developers.google.com/recaptcha/docs/verify

**Comment by PieterT2000** [comment]
You can also use the callback form of setState to get current state in callback handlers.

```
const [email, setEmail] = useState('')
const onVerify = response => {
  let newEmail;
  setEmail(currentState => {
    newEmail = currentState;
    return currentState;
  });
  console.log(newEmail);
}

**Comment by sarneeh** [comment]
It took a while but... fixed in `1.11.0` 😄
